File: javapatterns/src/main/java/org/iut/nantes/factories/cours/Salle.java
package org.iut.nantes.factories.cours;

abstract class Salle
{
  protected String type;
  protected int numero;

  public String getType() { return type; }
  public int getNumero() { return numero; }
}

class SalleInfo extends Salle
{
  public SalleInfo()
  {
    type = "salle info";
    numero = 42;
  }
}

class SalleMaths extends Salle
{
  public SalleMaths()
  {
    type = "salle classique";
    numero = 314;
  }
}

class SalleChimie extends Salle
{
  public SalleChimie()
  {
    type = "salle chimie";
    numero = 14;
  }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/factories/cours/Salle.kt
package org.iut.nantes.factories.cours

abstract class Salle {
    var type: String? = null
        protected set
    var numero = 0
        protected set
}

class SalleInfo : Salle() {
    init {
        type = "salle info"
        numero = 42
    }
}

class SalleMaths : Salle() {
    init {
        type = "salle classique"
        numero = 314
    }
}

class SalleChimie : Salle() {
    init {
        type = "salle chimie"
        numero = 14
    }
}